src/Illuminate/Session/Middleware/StartSession.php

```diff
@@ -53,12 +53,12 @@ public function handle($request, Closure $next)
 
         $this->collectGarbage($session);
 
-        $this->storeCurrentUrl($request, $session);
-
         $this->addCookieToResponse(
             $response = $next($request), $session
```

Comment:
Instead of writing this like so in which the next call gets lost in the other call this might be a more obvious approach:

```php
        $response = $next($request);

        $this->storeCurrentUrl($request, $session);
        $this->addCookieToResponse($response, $session);
```